runtime/lua/vim/treesitter.lua

```diff
@@ -664,4 +664,35 @@ function M.foldexpr(lnum)
   return require('vim.treesitter._fold').foldexpr(lnum)
 end
 
+---Get the commentstring for a buffer and position taking into account
+---language injections.
+---@param opts table|nil Optional keyword arguments:
+---             - bufnr integer|nil Buffer number (nil or 0 for current buffer)
+---             - pos table|nil 0-indexed (row, col) tuple. Defaults to cursor position in the
+---                             current window. Required if {bufnr} is not the current buffer
+---@return string
+function M.commentstring(opts)
+  opts = opts or {}
+
+  local bufnr = opts.bufnr
+  if not bufnr or bufnr == 0 then
+    bufnr = a.nvim_get_current_buf()
+  end
+
+  local parser = M.get_parser(opts.bufnr)
+  local range = resolve_pos(bufnr, opts.pos)
+  local commentstring
+  parser:for_each_child(function(tree, lang)
+    if tree:contains(range) then
+      local buf = a.nvim_create_buf(false, true)
+      vim.bo[buf].filetype = lang
```

Comment:
Won't this potentially be an issue because of the use of `lang`? For example in markdown, the lang may be `markdown_inline`, and it will then default to commentstring `/*%s*/`

Reply:
But markdown_inline will have an injection so we'd use that.

Reply:
Just define a commentstring for markdown_inline then. It hasn't a big problem.